runtime/development/src/lib.rs

```diff
@@ -1325,38 +1329,74 @@ impl xcm_executor::Config for XcmConfig {
 /// We need to ensure we have at least one rule per token we want to handle or else
 /// the xcm executor won't know how to charge fees for a transfer of said token.
 pub type Trader = (
-	FixedRateOfFungible<NativePerSecond, ()>,
-	FixedRateOfFungible<UsdPerSecond2000, ()>,
-	FixedRateOfFungible<UsdPerSecond3000, ()>,
+	FixedRateOfFungible<KsmPerSecond, ToTreasury>,
+	FixedRateOfFungible<NativePerSecond, ToTreasury>,
+	FixedRateOfFungible<KUsdPerSecond, ToTreasury>,
+	FixedRateOfFungible<UsdPerSecond, ToTreasury>,
+	FixedRateOfFungible<UsdPerSecondSibling, ToTreasury>,
 );
 
+pub struct ToTreasury;
+impl TakeRevenue for ToTreasury {
+	fn take_revenue(revenue: MultiAsset) {
+		use orml_traits::MultiCurrency;
+		use xcm_executor::traits::Convert;
+
+		if let MultiAsset {
+			id: Concrete(location),
+			fun: Fungible(amount),
+		} = revenue
+		{
+			if let Ok(currency_id) =
+				<CurrencyIdConvert as Convert<MultiLocation, CurrencyId>>::convert(location)
+			{
+				let _ = OrmlTokens::deposit(currency_id, &TreasuryAccount::get(), amount);
+			}
+		}
+	}
+}
+
 parameter_types! {
+	pub KsmPerSecond: (AssetId, u128) = (MultiLocation::parent().into(), ksm_per_second());
+
 	pub NativePerSecond: (AssetId, u128) = (
 		MultiLocation::new(
 			1,
-			X2(Parachain(2000), GeneralKey(CurrencyId::Native.encode())),
+			X2(Parachain(2088), GeneralKey(CurrencyId::Native.encode())),
```

Comment:
I remember that in the docs for XCM they stated that the `GeneralKey` variant should be omitted if possible. But I have seen this quite a lot also in other projects. Is this currently the way to go? Is there a global registry or is the key just for us?

Reply:
Do you have a link to that resource? As far as I know, Multilocations are canonical, so in some places such as the traders and the `CurrencyIdConvert` the specific para id and general key need to be used.

Reply:
https://github.com/paritytech/xcm-format -> search for GeneralKey

Reply:
Thanks for the link. Yeah, those docs are pretty abstract. This setup is similar to a lot of projects I have looked at, namely Acala.

>  Is there a global registry or is the key just for us?

There is no "global registry" per se, but rather a global knowledge of these keys. For instance, for Acala to support AIR, we need to tell them our para id and AIR's general key.